server/src/main/java/org/apache/iotdb/db/engine/merge/manage/MergeResource.java

```diff
@@ -263,4 +266,28 @@ public void setMeasurementSchemaMap(Map<PartialPath, MeasurementSchema> measurem
   public void clearChunkWriterCache() {
     this.chunkWriterCache.clear();
   }
+
+  public void updateStartTime(TsFileResource tsFileResource, String device, long startTime) {
+    Map<String, Pair<Long, Long>> deviceStartEndTimePairMap =
+        startEndTimeCache.getOrDefault(tsFileResource, new HashMap<>());
+    Pair<Long, Long> startEndTimePair =
+        deviceStartEndTimePairMap.getOrDefault(device, new Pair<>(Long.MAX_VALUE, Long.MIN_VALUE));
+    long newStartTime = startEndTimePair.left > startTime ? startTime : startEndTimePair.left;
+    deviceStartEndTimePairMap.put(device, new Pair<>(newStartTime, startEndTimePair.right));
+    startEndTimeCache.put(tsFileResource, deviceStartEndTimePairMap);
+  }
```

Comment:
Use `computeIfAbsent` then you may avoid calling extra `put` and you can directly modify the time pairs.